### ISSUE TITLE
Using the base option causes the manifest to use broken absolute paths

### ISSUE THREAD

**Issue by maurizi** [smysnk/gulp-rev-all#43]
When I use:
```javascript
    return gulp.src('assets/**')
         // Don't version source map files
        .pipe(revall({ ignore: [ /\.js\.map$/ ]}))
        .pipe(gulp.dest('static/'))
        .pipe(revall.manifest())
        .pipe(gulp.dest('static/'));
```
I get a correctly made `rev-manifest.json` with relative filenames as keys.

If I use:
```javascript
    return gulp.src('assets/js/**', { base: 'assets/' })
         // Don't version source map files
        .pipe(revall({ ignore: [ /\.js\.map$/ ]}))
        .pipe(gulp.dest('static/'))
        .pipe(revall.manifest())
        .pipe(gulp.dest('static/'));
```
The `rev-manifest.json` *tries* to use absolute paths as the file keys (which isn't great).  **However**, it cuts off the beginning of the path, so if my file is located at `/home/maurizi/project/assets/js/app.js`, what ends up in the `rev-manifest.json` is `"e/maurizi/project/assets/js/app.js": "/static/js/app.js"`

**Comment by vinnymac** [comment]
I think this is what was occurring in #55. I have the same experience where my paths are being trimmed after having set a base on my gulp.src. I'm actually having difficulty getting any of the base settings to work.

i.e. I have a folder at './dist/assets/' which is my base. I want my manifest to say something like './assets/app.css' However all I get in return is 'ssets/app.css'.

Best I can do is get the manifest to point directly at the file names and then I can modify in js to append the base myself.

Thanks

**Comment by island205** [comment]
+1

**Comment by derflocki** [comment]
+1

**Comment by derflocki** [comment]
I have discovered, that if i output the sources first thing, relative path's are written:

	return gulp.src('assets/js/**', { base: 'assets/' })
        .pipe(gulp.dest('static/'))
         // Don't version source map files
        .pipe(revall({ ignore: [ /\.js\.map$/ ]}))
        .pipe(gulp.dest('static/'))
        .pipe(revall.manifest())
        .pipe(gulp.dest('static/'));

**Comment by lynndylanhurley** [comment]
I'm having the same issue

I have a solution, but it's is crazy ugly. I'm using gulp-replace to modify the manifest before writing to disk. My base directory is called `app`. The regex deletes anything between the opening quotation (`"`) and the base directory name (`/app/`).

~~~javascript
// revise the files
var fingerprint = lazypipe()
  .pipe(-> $.revAll({
    ignore: [
      /index.html$/
    ]
  }))

// write the manifest to disk
var genManifest = lazypipe()
  .pipe(-> $.revAll.manifest())
  .pipe(-> $.replace(/[^"]*(\/app\/)/, ''))
  .pipe(-> gulp.dest(process.env.DIST_DIR))
~~~

This is not ideal, but it works for my use case.

**Comment by smysnk** [comment]
Should be fixed in v0.8.0, please give it a try.
